### PR TITLE
Support add u/this

### PR DESCRIPTION
**Pull request by Na-Nazhou**
Todo:
- [x] Current the validation for Url is too strict some website urls cannot pass it.
e.g.  https://www.google.com.sg/#spf=1571833141805
The path segment can contain just `//` 
image

- [x] Test for `add u/this`

### CODE REVIEW

**Comment by Na-Nazhou** [comment]
Closes #89

File: src/main/java/seedu/mark/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -38,19 +39,26 @@ public AddCommand parse(String args) throws ParseException {
...
         // compulsory fields
         Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-        Url url = ParserUtil.parseUrl(argMultimap.getValue(PREFIX_URL).get());
+        String urlString = argMultimap.getValue(PREFIX_URL).get();
+        Url url;
+        if (!"this".equals(urlString)) {
+            url = ParserUtil.parseUrl(urlString);
+        } else {
+            isMissingUrl = true;
+            url = new Url(BookmarkBuilder.DEFAULT_URL);
```
[the remark this answers is not in the document]
Comment (Na-Nazhou) [reply src/main/java/seedu/mark/logic/parser/AddCommandParser.java]: Omg I forgot that.

[the remark this answers is not in the document]
Comment (Na-Nazhou) [reply src/main/java/seedu/mark/logic/parser/AddCommandParser.java]: An exception will be thrown if there is no internet connection

File: src/main/java/seedu/mark/logic/commands/AddCommand.java [hunk 2]
```diff
@@ -35,21 +36,42 @@
...
     @Override
     public CommandResult execute(Model model, Storage storage) throws CommandException {
         requireAllNonNull(model, storage);
 
+        if (hasNoUrl) {
+            Url urlToAdd = model.getCurrentUrl();
+            if (urlToAdd == null) {
+                throw new CommandException(MESSAGE_MISSING_URL);
+            }
+            toAdd = new Bookmark(toAdd.getName(), urlToAdd, toAdd.getRemark(), toAdd.getFolder(),
```
Comment (TSAI-HSIAO-HAN) [inline src/main/java/seedu/mark/logic/commands/AddCommand.java]:
```suggestion
            toAdd = new BookmarkBuilder(toAdd).with(urlToAdd).build();
```
May be easier to read. And also need to import BookmarkBuilder.

**Pull request merged**